scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl

```diff
@@ -256,16 +262,48 @@ task ExtractFromSuperpartitionedTables {
                 ORDER BY location
             " --call_set_identifier ~{call_set_identifier} --dataset_name ~{dataset_name} --table_name vet_${str_table_index} --project_id ~{project_id}
 
-            python3 /app/run_avro_query.py --sql "
-                EXPORT DATA OPTIONS(
-                uri='${avro_prefix}/refs/ref_ranges_${str_table_index}/ref_ranges_${str_table_index}_*.avro', format='AVRO', compression='SNAPPY') AS
-                SELECT location, r.sample_id, length, state
-                FROM \`~{project_id}.~{dataset_name}.ref_ranges_${str_table_index}\` r
-                INNER JOIN \`~{project_id}.~{dataset_name}.sample_info\` s ON s.sample_id = r.sample_id
-                WHERE withdrawn IS NULL AND
-                is_control = false
-                ORDER BY location
-            " --call_set_identifier ~{call_set_identifier} --dataset_name ~{dataset_name} --table_name ref_ranges_${str_table_index} --project_id ~{project_id}
+            if [ ~{use_compressed_references} = false ]; then
+                python3 /app/run_avro_query.py --sql "
+                    EXPORT DATA OPTIONS(
+                    uri='${avro_prefix}/refs/ref_ranges_${str_table_index}/ref_ranges_${str_table_index}_*.avro', format='AVRO', compression='SNAPPY') AS
+                    SELECT location, r.sample_id, length, state
+                    FROM \`~{project_id}.~{dataset_name}.ref_ranges_${str_table_index}\` r
+                    INNER JOIN \`~{project_id}.~{dataset_name}.sample_info\` s ON s.sample_id = r.sample_id
+                    WHERE withdrawn IS NULL AND
+                    is_control = false
+                    ORDER BY location
+                " --call_set_identifier ~{call_set_identifier} --dataset_name ~{dataset_name} --table_name ref_ranges_${str_table_index} --project_id ~{project_id}
+            else
+                python3 /app/run_avro_query.py --sql "
+                    CREATE TEMP FUNCTION intToState(state INT64)
+                        RETURNS string
+                        AS (
+                          CASE state
+                            WHEN 7 THEN 'v'
+                            WHEN 8 THEN '*'
+                            WHEN 9 THEN 'm'
+                            WHEN 10 THEN 'u'
+                            ELSE CAST(state as string)
+                          END
+                    );
+                    CREATE TEMP FUNCTION UnpackRefRangeInfo(superpackEntry int64)
+                        RETURNS STRUCT<location INT64, len INT64, state string>
+                        AS (
+                          STRUCT(
+                          1000000000000 * ((superpackEntry >> 48) & 0xFFFF) + ((superpackEntry >> 16) & 0xFFFFFFFF),
+                          (superpackEntry >> 4) & 0xFFF,
+                          intToState(superpackEntry & 0xF))
+                    );
```

Comment:
Something I think I've heard @gbggrant ask for: can we have these functions made available in the GVS schema so ad-hoc querying does not become unpossible (or very unconvenient)?

Reply:
not in the first pass, although the SQL to do the queries are present in the tickets including examples of how to use them to make the content look like before.  I am MUCH more interested in getting the initial implementation in, given our timelines, than minimizing the developer effort.  In the meantime, when I do ad-hoc querying on tables, I just copy/paste the file of utility functions into my sql console and use them